lib/Ed25519Signature2020.js

```diff
@@ -0,0 +1,148 @@
+      if(typeof key.signer === 'function') {
+        this.signer = key.signer();
+      }
+      if(typeof key.verifier === 'function') {
+        this.verifier = key.verifier();
+      }
+    }
+  }
+
+  /**
+   * Adds a signature (proofValue) field to the proof object. Called by
+   * LinkedDataSignature.createProof().
+   *
+   * @param {object} options - The options to use.
+   * @param {Uint8Array} options.verifyData - Data to be signed.
+   * @param {object} options.proof - Proof object (containing the proofPurpose,
+   *   verificationMethod, etc).
+   *
+   * @returns {Promise<object>} Resolves with the proof containing the signature
+   *   value.
+   */
+  async sign({verifyData, proof}) {
+    if(!(this.signer && typeof this.signer.sign === 'function')) {
+      throw new Error('A signer API has not been specified.');
+    }
+    const signatureBytes = await this.signer.sign({data: verifyData});
+
+    // prefix with `z` to indicate multi-base base58btc encoding
+    proof.proofValue = {'@value': `z${base58btc.encode(signatureBytes)}`,
+      '@type': 'https://w3id.org/security#multibase'};
+    return proof;
+  }
+
+  /**
+   * Verifies the proof signature against the given data.
+   *
+   * @param {object} options - The options to use.
+   * @param {Uint8Array} options.verifyData - Canonicalized hashed data.
+   * @param {object} options.verificationMethod - Key object.
+   * @param {object} options.proof - The proof to be verified.
+   *
+   * @returns {Promise<boolean>} Resolves with the verification result.
+   */
+  async verifySignature({verifyData, verificationMethod, proof}) {
+    const {proofValue} = proof;
+    if(!(proofValue && typeof proofValue === 'string')) {
+      throw new TypeError(
+        'The proof does not include a valid "proofValue" property.');
+    }
+    if(proofValue[0] !== 'z') {
+      throw new Error('Only base58btc multibase encoding is supported.');
+    }
+    const signatureBytes = base58btc.decode(proofValue.substr(1));
+
+    let {verifier} = this;
+    if(!verifier) {
+      const key = await this.LDKeyClass.from(verificationMethod);
+      verifier = key.verifier();
+    }
+    return verifier.verify({data: verifyData, signature: signatureBytes});
+  }
+
+  async assertVerificationMethod({verificationMethod}) {
+    if(!jsonld.hasValue(verificationMethod, 'type', this.requiredKeyType)) {
+      throw new Error(
+        `Invalid key type. Key type must be "${this.requiredKeyType}".`);
+    }
+  }
+
+  async getVerificationMethod({proof, documentLoader}) {
+    if(this.key) {
+      return this.key.export({publicKey: true});
+    }
+
+    const verificationMethod = await super.getVerificationMethod(
+      {proof, documentLoader});
+    await this.assertVerificationMethod({verificationMethod});
+    return verificationMethod;
+  }
+
+  async matchProof({proof, document, purpose, documentLoader, expansionMap}) {
+    if(!await super.matchProof({proof, document, purpose, documentLoader,
```

Comment:
Check the document's `@context` to ensure it includes the ed25519-2020 context -- and if not, there's no match.

Reply:
Added